src/Idris/ParseExpr.hs

```diff
@@ -452,14 +452,19 @@ bracketed' open syn =
             do (FC f start (l, c)) <- getFC
                lchar ')'
                return $ PTrue (spanFC open (FC f start (l, c+1))) TypeOrTerm
-        <|> try (do (ln, lnfc) <- name
-                    colonFC <- lcharFC ':'
-                    lty <- expr syn
-                    starsFC <- reservedOpFC "**"
+        <|> try (do (ln, lnfc, colonFC, lty, starsFC):bnds <- some $ do
+                        (ln, lnfc, colonFC) <- try $ do
+                          (ln, lnfc) <- name
+                          colonFC <- lcharFC ':'
```

Comment:
Basically, it expects that we get a list (a : ty1 ** a : ty2 ** .. ). It should be possible to make the type optional, but I am concerned about ambiguity of the expression.